### PR TITLE
Adding jobs summary

### PR DESCRIPTION
**Pull request by beatrizsanchezt**
https://github.com/ndiaz-candido/jobs-ui/issues/1

### CODE REVIEW

File: components/JobsSummary.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  },
+  title: {
+    fontSize: 14,
+  },
+  pos: {
+    marginBottom: 10,
+  },
+});
+
+function JobSummary({ data }) {
```
Comment (ndiaz-candido) [inline components/JobsSummary.js:28]: The file is called `JobsSummary` but the component is called `JobSummary`

File: components/JobsSummary.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        >
+          Apply
+        </Button>
+      </CardActions>
+    </Card>
+  );
+}
+
+JobSummary.propTypes = {
+  data: PropTypes.string.isRequired,
```
Comment (ndiaz-candido) [inline components/JobsSummary.js:64]: It's always better to pass the properties without a object wrapper. I mean, here you're only getting a `data` prop but inside it you have `role`, `company`, `location`, etc. It's safer to check each one. Something like:
```js
JobSummary.propTypes = {
  role: PropTypes.string,
  company: PropTypes.string
};
```

File: components/layout.js [hunk 3]
```diff
@@ -1,4 +1,6 @@
+import React from 'react';
```
Comment (ndiaz-candido) [inline components/layout.js:1]: I noticed you're using capital case for naming files. Could you rename this file to `Layout`?

File: pages/api/jobs.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  {
+    "role": "Product Owner",
+    "company": "FIS",
+    "location": "Cambridge, London",
+    "summary": "Experienced (relevant combo of work and education). Provide the internal voice of the customer for the team and work with Product Managers, Customers (as well…",
+    "founded-date": "¡30 days ago",
+    "link": "https://fis/jobs"
+  }
+]
```
Comment (ndiaz-candido) [inline pages/api/jobs.json:34]: EOF

File: pages/index.js [hunk 5]
```diff
@@ -1,30 +1,53 @@
...
-            <ListItem>Job1</ListItem>
-            <ListItem>Job2</ListItem>
-            <ListItem>Job3</ListItem>
-            <ListItem>Job4</ListItem>
-          </List>
-        </Container>
-      </div>
+      <Grid container className={classes.root} spacing={2}>
+        <Grid item xs={6}>
+          <Container maxWidth="sm">
```
Comment (ndiaz-candido) [inline pages/index.js:23]: Could you define a `JobsSummaryList` component? and move the `map` to there :)

File: pages/index.js [hunk 6]
```diff
@@ -1,30 +1,53 @@
...
-  jobsContent: {
-    padding: theme.spacing(8, 0, 6),
+const useStyles = makeStyles(() => ({
+  root: {
+    padding: '0 30px',
   },
 }));
 
-export default () => {
+function Index({ data }) {
```
Comment (ndiaz-candido) [inline pages/index.js:17]: Same comment as in `JobsSummary` about checking each prop :)

File: pages/index.js [hunk 7]
```diff
@@ -1,30 +1,53 @@
...
+        <Grid item xs={6}>
+          <JobDescription />
+        </Grid>
+      </Grid>
     </Layout>
   );
+}
+
+export async function getServerSideProps() {
+  const response = await fetch(`http://localhost:3000/api/jobs`);
```
Comment (ndiaz-candido) [inline pages/index.js:38]: I think just defining `/api/jobs` is enough here. If you include the whole host it could be problem in production.

Comment (beatrizsanchezt) [reply pages/index.js]: Pending to add URL base on the environment.

File: pages/index.js [hunk 8]
```diff
@@ -1,30 +1,53 @@
...
+
+Index.propTypes = {
+  jobObject: PropTypes.shape({
+    role: PropTypes.string.isRequired,
+    company: PropTypes.string.isRequired,
+    location: PropTypes.string.isRequired,
+    summary: PropTypes.string.isRequired,
+    link: PropTypes.string.isRequired,
+  }).isRequired,
+  jobs: PropTypes.arrayOf(PropTypes.jobObject).isRequired,
```
Comment (ndiaz-candido) [inline pages/index.js:50]: `jobObject` is not necessary, include the shape as follows:

```js
jobs: PropTypes.arrayOf(PropTypes.shape({
    role: PropTypes.string.isRequired,
    company: PropTypes.string.isRequired,
    location: PropTypes.string.isRequired,
    summary: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
  })).isRequired
```

File: components/layout.js [hunk 9]
```diff
@@ -29,9 +30,15 @@ const Layout = ({ title, children }) => {
...
       </AppBar>
+      <Toolbar />
       {children}
     </div>
   );
 };
 
+Layout.propTypes = {
+  title: PropTypes.string.isRequired,
+  children: PropTypes.element.isRequired,
```
Comment (ndiaz-candido) [inline components/layout.js:41]: `PropTypes.node.isRequired` fits better

File: components/JobsSummaryList.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+          summary={item.summary}
+          link={item.link}
+        />
+      ))}
+    </Container>
+  );
+}
+
+JobsSummaryList.propTypes = {
+  jobs: PropTypes.arrayOf(PropTypes.object).isRequired,
```
Comment (ndiaz-candido) [inline components/JobsSummaryList.js:24]:
```js
jobs: PropTypes.arrayOf(PropTypes.shape({
    role: PropTypes.string.isRequired,
    company: PropTypes.string.isRequired,
    location: PropTypes.string.isRequired,
    summary: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
  })).isRequired
```

Comment (ndiaz-candido) [reply components/JobsSummaryList.js:24]: One more, jobs can be optional, this component can be rendered event with no jobs. Could you remove the `isRequired` and add:

```js
JobsSummaryList.defaultProps = {
  jobs: []
}
```

File: pages/index.js [hunk 11]
```diff
@@ -1,30 +1,53 @@
...
+import JobsSummaryList from '../components/JobsSummaryList';
+import JobDescription from '../components/JobDescription';
+import Layout from '../components/Layout';
 
-const useStyles = makeStyles((theme) => ({
-  jobsContent: {
-    padding: theme.spacing(8, 0, 6),
+const useStyles = makeStyles(() => ({
+  root: {
+    padding: '0 30px',
```
Comment (ndiaz-candido) [inline pages/index.js:12]: Use `rem` instead of `px` for paddings

File: pages/index.js [hunk 12]
```diff
@@ -1,30 +1,53 @@
...
+        <Grid item xs={6}>
+          <JobDescription />
+        </Grid>
+      </Grid>
     </Layout>
   );
+}
+
+export async function getServerSideProps() {
+  const response = await getJobsSummary();
```
Comment (ndiaz-candido) [inline pages/index.js:33]:
```js
  const { jobs, status } = await getJobsSummary();
```

File: pages/index.js [hunk 13]
```diff
@@ -1,30 +1,53 @@
...
+  return {
+    props: {
```
Comment (ndiaz-candido) [inline pages/index.js:35]:
```js
props: { jobs, status }
```

**Comment by ndiaz-candido** [comment]
@beatrizsanchezt is this PR still needed?

**Pull request closed without merging**